student.rb

```diff
@@ -1,14 +1,22 @@
 require './person'
+require './classroom'
 
 class Student < Person
-  attr_accessor :specialization
+  attr_accessor :specialization, :classroom
 
   def initialize(classroom, age, name = 'Unknown', parent_permission: 'Unknown')
-    super(age, name, parent_permission)
+    raise "The class doesn't exist, please create an object classroom first" unless classroom.is_a?(Classroom)
+
+    super(age, name, parent_permission: parent_permission)
     @classroom = classroom
+    @classroom.students << self
   end
```

Comment:
- [ Required ] It would be better to have a `setter` method that is dedicated to setting the classroom for a `student` rather than setting it only in the class constructor. Setting it in the constructor and not availing a method to set the `classroom` makes it impossible to set the `student's` classroom after the student has been created with a `nil`. You can achieve this by refactoring your class `Student` code as shown below

```ruby
require './person'
require './classroom'

class Student < Person
  attr_accessor :specialization, :classroom

  def initialize(classroom, age, name = 'Unknown', parent_permission: 'Unknown')
    super(age, name, parent_permission: parent_permission)
    @classroom = classroom
  end

 # Setter method
  def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
 end

  def play_hooky
    '¯(ツ)/¯'
  end

  def class?
    @classroom.label
  end
end

```

Reply:
I did it 🎉 

I also tried to raise an error when a user tries to register the same student two times in the same classroom. Do you think it's well implemented?

Reply:
The `classroom setter` method takes care of that.

Reply:
Thanks for the response 😁